test/action_view/invalid_components_test.rb

```diff
@@ -1,11 +1,5 @@
 # frozen_string_literal: true
 
-require "invalid_components_test_helper"
```

Comment:
neat! It's nice we can have invalid components in `app/components` now 😄 and that the app wont complain until you try to render the component.

Reply:
I'm torn as to whether this is a good thing or not. Can you elaborate why you think it's good that the app doesn't raise an error until you try to render the component, vs. when it compiles the component on boot, which only happens when eager loading is turned on?

Reply:
I'm torn too. But in general Rails is lazy about a bunch of things like this and/or as a developer you only figure you broke something when you 'run it' (and lets hope you run it in your test suite). For example
1. I forget to add a view for a controller action
2. The table backing a model is missing/typoed
3. I've typoed the name of some ActiveRecored relation and AR can't find the association by convention
etc. etc.
Its fair to say that some (maybe all) of those cases are hard for Rails to check for at eager load time but that doesn't detract from the fact that those kinds of surprises are not unusual in Rails in general. 

On the flip side I worry a bit that failing at eager load time could be really disruptive. Suppose I'm in the middle of some refactor, I'm moving files around, renaming components, I haven't moved/renamed the views yet because I'm focused on the Component API but now I can't even load the app or maybe run my test suite? That feels pretty ugly.

Lastly, I think we should be consistent and treat all of these 'template error' cases the same way:

1. no initialize
1. no views at all
1. no default view (is this ok if we have variants?)
1. more than one default view
1. to many variant views for the same variant

Note that we currently treat 1 differently than 2, 4 & 5.

Reply:
@bostanio 💥 you make a very strong point here.

We've always tried to have this library behave consistently with Rails. With that goal in mind, I think we're right to make this change. 

Thank you for taking the time to explain your perspective ❤️

Reply:
Sorry for not getting back yesterday.

@bostanio did a great job of making the case for this. Like he said, my thinking was that it's nice to match how Rails doesn't blow up when you forget to write a view for a controller until you call `render`. Which feels less disruptive when you're adding/refactoring a component.

But it can also be helpful to see those errors sooner, so 🤷‍♂I could see an argument for wanting it the other way. But I think the consistency with Rails is valuable, so I'd vote for not blowing up on eager load.